### ISSUE TITLE
Broken migration from 9.1.2. South migrations missing.

### ISSUE THREAD

**Issue by shulcsm** [getsentry/onpremise#591]
Hello,

There are bunch of south migrations that are not present in 9.1.2 tag. I assume some of them are datamigrations since i'm getting integrity error about null values in `sentry_groupedmessage.project_id` when upgrade calls `sentry.0002_912_to_recent`.

This commit https://github.com/getsentry/sentry/commit/3b77b756044db4547a8d55490d48ae5b43599031 removed south.
As you can see last migration in there is `0518` but last migration tagged in 9.1.2 is `0472`.
I'm not 100% sure that this solves my upgrade issue but shouldn't these missing changes be cherrypicked/backported?

**Comment by BYK** [comment]
Can you share your install process and logs so we can investigate? What's your upgrade path? (from which version are you coming to and to which version you are upgrading to)

**Comment by wedamija** [comment]
> @BYK
> Install process is just calling ./install.sh
> Tried upgrading to 10.0.0 and whatever latest is with onpremise HEAD.
> Here is the relevant log part: https://gist.github.com/shulcsm/eae055713b576cdd650b20e56d84e06f
> My db schema is valid for 9.1.2, but data seems broken.
> 
> I found [getsentry/sentry#14572](https://github.com/getsentry/sentry/pull/14572) but that migration (including bunch of others) is not part of 9.1.2 yet 10.0.0 expects it applied

It's intended that we didn't include migrations after 0472 - They're collapsed and included in `sentry.0002_912_to_recent`. The migration in https://github.com/getsentry/sentry/pull/14572 is actually included, you can see it here: https://github.com/getsentry/sentry/blob/0765032e8019ed840dbf3ffbc7e11534ad142f4c/src/sentry/migrations/0002_912_to_recent.py#L684 (not obvious I know, but basically that's changing the field from not null to null).

I don't think we ever wrote a data migration to fix `Group` rows that have no project id. Based on what I see in 0501, all that migration does is set the column not null, and would have also failed for your install. I'm guessing that the assumption was that since we had no invalid rows in our hosted Sentry database then it'd be fine.

Could you run `select count(*) from sentry_groupedmessage where project_id is null` so that we can get an idea of how many rows are causing this issue? I think we'll likely just want to delete any groups that have no project_id, since I don't know how they would even be accessible.

**Comment by shulcsm** [comment]
Sorry about confusion with migrations. 
There where only 50 rows in sentry_groupedmessage so i deleted them.
Now i'm hitting this: 
https://gist.github.com/shulcsm/82d8f704b38f4f4edb3a2c0be154a0d2

**Comment by BYK** [comment]
Okay, this one looks like the same issue with #468. I'll investigate. Thanks for sharing and for your patience.

**Comment by shulcsm** [comment]
Thank you for your time. If it helps problematic events should be from  8.13
#468 is indeed same I think you can close this issue in favor of that.

**Comment by BYK** [comment]
>If it helps problematic events should be from 8.13

This confirms @wedamija's suspicions, thanks for sharing :)

>#468 is indeed same I think you can close this issue in favor of that.

I'd like to keep this one open for that initial issue as we may be able to incorporate a fix into the install script. What did you do to delete these? Manual SQL queries or from django/sentry shell?

**Comment by shulcsm** [comment]
Yes, manual queries:
```sql
DELETE FROM sentry_groupedmessage WHERE project_id IS NULL;
DELETE FROM sentry_message WHERE group_id NOT in (SELECT id FROM sentry_groupedmessage);
```
I'm surprised there were no foreign key constraints between the two.

**Comment by BYK** [comment]
Closing as this should now be fixed with #468.
